File: app/assets/javascripts/models/tournament.js
TournyMadness.Tournament = DS.Model.extend({
  name: DS.attr('string'),
  description: DS.attr('string'),
  start_date: DS.attr('date'),
  end_date: DS.attr('date'),
  sign_up_start: DS.attr('date'),
  sign_up_end: DS.attr('date'),
  style: DS.attr('string'),
  champion: DS.attr('string'),
  teams: DS.hasMany('team'),
  games: DS.hasMany('game')
});

File: app/models/tournament.rb
class Tournament < ActiveRecord::Base
  has_many :teams, inverse_of: :tournament, dependent: :destroy
  has_many :games, inverse_of: :tournament, dependent: :destroy

  validates :name, presence: true
  validates :start_date, presence: true
  validates :end_date, presence: true
end
